packages/botonic-plugin-contentful/src/contentful/cms-contentful.ts

```diff
@@ -0,0 +1,229 @@
+import { ScheduleDelivery } from './contents/schedule'
+import { DateRangeDelivery } from './contents/date-range'
+import { ImageDelivery } from './contents/image'
+import { AssetDelivery } from './contents/asset'
+import { QueueDelivery } from './contents/queue'
+import { ButtonDelivery } from './contents/button'
+import { ContentfulOptions } from '../plugin'
+import { CachedClientApi } from './delivery/cache'
+import { IgnoreFallbackDecorator } from './ignore-fallback-decorator'
+import { FollowUpDelivery } from './contents/follow-up'
+import * as contentful from 'contentful'
+import { SearchCandidate } from '../search'
+import { ContentsDelivery } from './contents/contents'
+
+export class Contentful implements cms.CMS {
+  private readonly _delivery: DeliveryApi
+  private readonly _contents: ContentsDelivery
+  private readonly _carousel: CarouselDelivery
+  private readonly _text: TextDelivery
+  private readonly _startUp: StartUpDelivery
+  private readonly _url: UrlDelivery
+  private readonly _keywords: KeywordsDelivery
+  private readonly _schedule: ScheduleDelivery
+  private readonly _dateRange: DateRangeDelivery
+  private readonly _image: ImageDelivery
+  private readonly _asset: AssetDelivery
+  private readonly _queue: QueueDelivery
+  private readonly _button: ButtonDelivery
+
+  /**
+   *
+   * See https://www.contentful.com/developers/docs/references/content-delivery-api/#/introduction/api-rate-limits
+   * for API rate limits
+   *
+   *  https://www.contentful.com/developers/docs/javascript/tutorials/using-js-cda-sdk/
+   */
+  constructor(options: ContentfulOptions) {
+    const client = createContentfulClientApi(options)
+    const deliveryApi = new AdaptorDeliveryApi(
+      options.disableCache
+        ? client
+        : new CachedClientApi(client, options.cacheTtlMs)
+    )
+    const delivery = new IgnoreFallbackDecorator(deliveryApi)
+    this._contents = new ContentsDelivery(delivery)
+
+    this._delivery = delivery
+    this._button = new ButtonDelivery(delivery)
+    this._carousel = new CarouselDelivery(delivery, this._button)
+    this._text = new TextDelivery(delivery, this._button)
+    this._startUp = new StartUpDelivery(delivery, this._button)
+    this._url = new UrlDelivery(delivery)
+    this._image = new ImageDelivery(delivery)
+    this._asset = new AssetDelivery(delivery)
+    this._queue = new QueueDelivery(delivery)
+    const followUp = new FollowUpDelivery(
+      this._delivery,
+      this._carousel,
+      this._text,
+      this._image,
+      this._startUp
+    )
+    ;[
+      this._text,
+      this._url,
+      this._carousel,
+      this._image,
+      this._startUp,
+    ].forEach(d => d.setFollowUp(followUp))
+    this._keywords = new KeywordsDelivery(delivery)
+    this._schedule = new ScheduleDelivery(delivery)
+    this._dateRange = new DateRangeDelivery(delivery)
+  }
+
+  button(id: string, context: Context = DEFAULT_CONTEXT): Promise<cms.Button> {
```

Comment:
This function and `element, image, chitchat` should also be "async"?

Reply:
there's an eslint rule which complain if you add async when there's no await.
async is only required when there's an await